packages/component-base/src/data-provider-controller/data-provider-controller.d.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import type { ReactiveController } from 'lit';
+import type { Cache } from './cache.js';
+
+type DataProviderParams = {
+  page: number;
+  pageSize: number;
+  parentItem?: unknown;
+  [key: string]: unknown;
```

Comment:
Instead of any arbitrary object, would be nice if the additional params could be provided explicitly as a generic type. Something like:

<img width="510" alt="Screenshot 2023-09-25 at 11 22 36" src="https://github.com/vaadin/web-components/assets/1222264/d2f2cff9-1862-47b5-89ec-8f5630cca1b1">

<img width="455" alt="Screenshot 2023-09-25 at 11 20 39" src="https://github.com/vaadin/web-components/assets/1222264/3dcbde4b-f02c-443c-a34b-8433fc9ecf81">

Reply:
Added.